osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs

```diff
@@ -38,28 +38,41 @@ public class OsuModRelax : ModRelax, IUpdatableByPlayfield, IApplicableToDrawabl
         private ReplayState<OsuAction> state = null!;
         private double lastStateChangeTime;
 
+        private DrawableOsuRuleset ruleset = null!;
+        private PressHandler pressHandler = null!;
+
         private bool hasReplay;
+        private bool legacyReplay;
 
         public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
         {
+            ruleset = (DrawableOsuRuleset)drawableRuleset;
+
             // grab the input manager for future use.
-            osuInputManager = ((DrawableOsuRuleset)drawableRuleset).KeyBindingInputManager;
+            osuInputManager = ruleset.KeyBindingInputManager;
         }
 
         public void ApplyToPlayer(Player player)
         {
             if (osuInputManager.ReplayInputHandler != null)
             {
                 hasReplay = true;
+
+                Debug.Assert(ruleset.ReplayScore != null);
+                legacyReplay = ruleset.ReplayScore.ScoreInfo.IsLegacyScore;
+
+                pressHandler = new LegacyReplayPressHandler(this);
```

Comment:
Should this still be a `LegacyReplayPressHandler` even for lazer-sourced (`legacyReplay == false`) replays?

Reply:
ah, nope. somehow didn't cause an issue when i tested